client/landing/gutenboarding/components/domain-picker/button.tsx

```diff
@@ -44,9 +48,16 @@ const DomainPickerButton: FunctionComponent = () => {
 		[ search, siteVertical ]
 	);
 
-	// Free .wordpress.com subdomain at index 0,
-	// Best paid domain match at index 1.
-	const [ freeDomainSuggestion, ...paidDomainSuggestions ] = suggestions ?? [];
+	const [ freeDomainSuggestions, paidDomainSuggestions ] =
+		suggestions?.reduce< [ DomainSuggestion[], DomainSuggestion[] ] >(
```

Comment:
Why not use [`_.partition()`](https://lodash.com/docs/4.17.15#partition)?

Reply:
I stayed away from it in the interest of not pulling in `_` for what's pretty easy to do, but after rewriting… it's very nice. Let's go with `_.partition`.